### ISSUE TITLE
MPI_ABORT invoked at the end of test run with failures

### ISSUE THREAD

**Issue by quetric** [Xilinx/ACCL#104]
Observed on a test against the emulator, with 8 ranks. All tests run, some fail, test killed at the very end with:
```
[1,7]<stdout>:3 tests failed on rank 7 (skipped 1 tests).
[1,6]<stdout>:3 tests failed on rank 6 (skipped 1 tests).
[1,2]<stdout>:3 tests failed on rank 2 (skipped 1 tests).
[1,3]<stdout>:3 tests failed on rank 3 (skipped 1 tests).
[1,4]<stdout>:3 tests failed on rank 4 (skipped 1 tests).
[1,1]<stdout>:3 tests failed on rank 1 (skipped 1 tests).
[1,5]<stdout>:3 tests failed on rank 5 (skipped 1 tests).
[1,0]<stdout>:3 tests failed on rank 0 (skipped 1 tests).
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 1 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
```

**Comment by Mellich** [comment]
This is explicitly called in our test suite if tests fail [here](https://github.com/Xilinx/ACCL/blob/cd1be8ec1e6f1fa8650ef8339a62d7c50e1099a4/test/host/xrt/test.cpp#L1417). I guess the idea is to make it easy to see that something went wrong. We may consider to change this and return a non-zero exit value instead.

**Comment by quetric** [comment]
this seems too heavy handed. We already report which tests failed, killing MPI only confuses things. I'll remove the abort

**Comment by quetric** [comment]
Fixes in 9d833db21da6 by using the number of failed tests as exit code, instead of invoking MPI abort
